cpp/ql/src/semmle/code/cpp/ir/dataflow/DefaultTaintTracking.qll

```diff
@@ -545,6 +545,9 @@ module TaintedWithPath {
     /** Override this to specify which elements are sinks in this configuration. */
     abstract predicate isSink(Element e);
 
+    /** Override this to specify additional barriers in this configuration. */
+    predicate isAdditionalBarrier(Expr node) { none() }
```

Comment:
Is there a reason why we shouldn't just stick to the names in the dataflow configurations and call this predicate `isBarrier`?

Reply:
As written `isAdditionalBarrier` isn't expected to include the default cases (from `nodeIsBarrier`) so it's not the same thing as an `isBarrier`.  That said, I think I should rewrite it so that it does work the same as `isBarrier` (with a call to the superclass to include the default barriers).

Reply:
Ah, it would have to work on `DataFlow::Node`s rather than `Expr`s or `Element`s if it's to handle the default cases correctly.  I'm reluctant because that would be inconsistent with the way `isSource` and `isSink` work.  I'm open to (concrete) suggestions.

Reply:
Can't we do something like:
```codeql
predicate isBarrier(Expr e) { nodeIsBarrier(getNodeForExpr(e)) }
```

I must admit that I didn't think it fully through yet. But that seems to be the way we go from expressions to nodes in the rest of `DefaultTaintTracking`.

Reply:
Yes, that works.  It does assume that all default barriers have a corresponding `Expr`, otherwise in principle we could be losing some.  I can't tell whether that's true from the code (but I'll run some tests).

Reply:
I ran lots of tests (locally) and saw no problems.  Just pushed the cleaned up version.